commands/wiki.js

```diff
@@ -1,48 +1,61 @@
+    levenshteinDistance(t.substr(1), s) + 1,
+    levenshteinDistance(s.substr(1), t.substr(1)) + (s[0] !== t[0] ? 1 : 0)
+  ) + 1;
+}
 
 const Module = new Augur.Module()
-.addCommand({name: "wiki",
-  description: "Search Wikipedia for a term.",
+.addCommand({
+  name: "wiki",
+  description: "Search the LDSG Wiki for a term.",
   syntax: "Term",
-  permissions: (msg) => (!msg.guild || (msg.channel.permissionsFor(msg.member).has(["EMBED_LINKS", "ATTACH_FILES"]) && msg.channel.permissionsFor(msg.client.user).has("ATTACH_FILES"))),
   process: async (msg, suffix) => {
-    try {
-      if (!suffix) return msg.reply("you need to tell me what to search!").then(u.clean);
-      const pf = new profanityFilter();
-      if (pf.scan(suffix.toLowerCase()).length == 0) {
-        let wiki = JSON.parse(await request(`https://en.wikipedia.org/w/api.php?action=opensearch&format=json&redirects=resolve&search=${encodeURIComponent(suffix)}`));
-
-        if (wiki && wiki[1].length > 0) {
-          let embed = u.embed()
-          .setTitle(wiki[1][0])
-          .setDescription(wiki[2][0])
-          .setURL(wiki[3][0])
-          .setAuthor("Wikipedia", "https://upload.wikimedia.org/wikipedia/commons/thumb/8/80/Wikipedia-logo-v2.svg/103px-Wikipedia-logo-v2.svg.png");
-
-          if (wiki[2][0].endsWith("may refer to:")) {
-            for (let i = 1; i < wiki[1].length; i++) {
-              embed.addField(wiki[1][i], wiki[2][i] + `\n[[Link]](${wiki[3][i]})`);
-            }
-          } else {
-            let body = await request(wiki[3][0]);
-            let $ = cheerio.load(body);
-
-            let img = $("table.infobox img");
-            if (img.length > 0) {
-              embed.setThumbnail("https:" + img.first().attr("src"));
-            }
-          }
-          msg.channel.send({embed});
-        } else {
-          msg.reply(`I couldn't find a result for \`${suffix}\` on <https://en.wikipedia.org>.`);
-        }
-      } else msg.reply("I'm not going to search for that. :rolling_eyes:").then(u.clean);
-    } catch(e) { u.errorHandler(e, msg); }
+    let items = (await getPage("shelves").catch((err) => { return { data: [] }; })).data
+      .concat((await getPage("books").catch((err) => { return { data: [] }; })).data);
+
+    // There are already more than 3 items, so if it's less than 3 something is broken
+    if (items.length < 3) {
+      msg.channel.send("I was unable to see the wiki properly.");
+      return;
+    }
+
+    items.sort((a, b) => {
+      let aDistance = levenshteinDistance(a.name, suffix);
+      let bDistance = levenshteinDistance(b.name, suffix);
+      return aDistance > bDistance ? 1 : -1;
+    });
+
+    let embed = u.embed()
+      .setTitle("Search results for `" + suffix + "`")
+      .addFields(
+        { name: items[0].name, value: items[0].description },
+        { name: items[1].name, value: items[1].description },
+        { name: items[2].name, value: items[2].description }
+      );
```

Comment:
I'm doing some silly stuff here. Don't judge.
```suggestion
    items = items.sort((a, b) => {
      let aDistance = levenshteinDistance(a.name, suffix);
      let bDistance = levenshteinDistance(b.name, suffix);
      return aDistance > bDistance ? 1 : -1;
    }).splice(0, 3).map(item => ({name: item.name, value: item.description}));
    // Note to add a link once we have a good way to *get* one.

    let embed = u.embed()
      .setTitle("Search results for `" + suffix + "`")
      .addFields(items);
```